### PR TITLE
Adding kube client for MCS API

### PR DESCRIPTION
**Pull request by nmittler**
Split out from #31401

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: manifests/charts/base/files/gen-istio-cluster.yaml [hunk 1]
```diff
@@ -6012,6 +6012,11 @@ rules:
   - apiGroups: [""]
     resources: ["secrets"]
     verbs: ["get", "watch", "list"]
+
+  # Used for MCS serviceexport management
+  - apiGroups: ["multicluster.x-k8s.io"]
+    resources: ["serviceexports"]
+    verbs: ["create", "delete"]
```
Comment (howardjohn) [inline manifests/charts/base/files/gen-istio-cluster.yaml:6019]: are you sure we only want these? and not get/watch/list as well?

**Review by howardjohn — APPROVED** [review]

File: manifests/charts/base/files/gen-istio-cluster.yaml (hunk 1 above)
Comment (nmittler) [reply manifests/charts/base/files/gen-istio-cluster.yaml:6019]: IIRC that was a comment on the original PR as well. I have no problem with that ... probably makes sense.

**Comment by josephpeacock** [comment]
@googlebot I consent.

File: manifests/charts/base/files/gen-istio-cluster.yaml (hunk 1 above)
Comment (josephpeacock) [reply manifests/charts/base/files/gen-istio-cluster.yaml]: I can add some context there - this is the minimal set of permissions for the ServiceExport functionality to work. Eventually, we'll need get/watch/list to support ServiceImport, but I wanted to add the smallest set of access I could. I see no problem expanding it prematurely though, as we know ServiceImport will happen soon

**Pull request merged**